flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphRestartTest.java

```diff
@@ -742,41 +704,32 @@ public void testFailureWhileRestarting() throws Exception {
 
 	private Scheduler createSchedulerWithInstances(int num, TaskManagerGateway taskManagerGateway) {
```

Comment:
The last issue. We cannot replace it simply with `new TestingLogicalSlotProvider(num, gateway)` because the test based on it tests slot sharing. And for test coverage, we need a testing slot provider that implements slot sharing semantic.